cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,574 @@
+			Usage: "Print the labels of the pod",
+		},
+		cli.BoolFlag{
+			Name:  "no-trunc",
+			Usage: "Display the extended information",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "Print the numeric IDs of the pods only",
+		},
+		cli.StringFlag{
+			Name:  "sort",
+			Usage: "Sort output by id, name, or number",
+			Value: "name",
+		},
+	}
+	podPsDescription = "Prints out information about pods"
+	podPsCommand     = cli.Command{
+		Name:                   "ps",
+		Aliases:                []string{"ls", "list"},
+		Usage:                  "List pods",
+		Description:            podPsDescription,
+		Flags:                  podPsFlags,
+		Action:                 podPsCmd,
+		UseShortOptionHandling: true,
+	}
+)
+
+func podPsCmd(c *cli.Context) error {
+	if err := validateFlags(c, podPsFlags); err != nil {
+		return err
+	}
+
+	if err := podCheckFlagsPassed(c); err != nil {
+		return errors.Wrapf(err, "error with flags passed")
+	}
+
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "error creating libpod runtime")
+	}
+
+	defer runtime.Shutdown(false)
+
+	if len(c.Args()) > 0 {
+		return errors.Errorf("too many arguments, ps takes no arguments")
+	}
+
+	format := genPodPsFormat(c.String("format"), c.Bool("quiet"), c.Bool("cgroup"), c.Bool("labels"), c.Bool("ctr-names"), c.Bool("ctr-ids"), c.Bool("ctr-status"))
+
+	opts := podPsOptions{
+		Filter:             c.String("filter"),
+		Format:             format,
+		NoTrunc:            c.Bool("no-trunc"),
+		Quiet:              c.Bool("quiet"),
+		Sort:               c.String("sort"),
+		IdsOfContainers:    c.Bool("ctr-ids"),
+		NamesOfContainers:  c.Bool("ctr-names"),
+		StatusOfContainers: c.Bool("ctr-status"),
+	}
+
+	var filterFuncs []libpod.PodFilter
+	if opts.Filter != "" {
+		filters := strings.Split(opts.Filter, ",")
+		for _, f := range filters {
+			filterSplit := strings.Split(f, "=")
+			if len(filterSplit) < 2 {
+				return errors.Errorf("filter input must be in the form of filter=value: %s is invalid", f)
+			}
+			generatedFunc, err := generatePodFilterFuncs(filterSplit[0], filterSplit[1], runtime)
+			if err != nil {
+				return errors.Wrapf(err, "invalid filter")
+			}
+			filterFuncs = append(filterFuncs, generatedFunc)
+		}
+	}
+
+	pods, err := runtime.Pods(filterFuncs...)
+	if err != nil {
+		return err
+	}
+
+	return generatePodPsOutput(pods, opts, runtime)
+}
+
+// podCheckFlagsPassed checks if mutually exclusive flags are passed together
+func podCheckFlagsPassed(c *cli.Context) error {
+	// quiet, and format with Go template are mutually exclusive
+	flags := 0
+	if c.Bool("quiet") {
+		flags++
+	}
+	if c.IsSet("format") && c.String("format") != formats.JSONString {
+		flags++
+	}
+	if flags > 1 {
+		return errors.Errorf("quiet, and format with Go template are mutually exclusive")
+	}
+	return nil
+}
+func generatePodFilterFuncs(filter, filterValue string, runtime *libpod.Runtime) (func(pod *libpod.Pod) bool, error) {
```

Comment:
We should avoid this approach for now - honestly filter functions in the API were kind of a mistake (at least as implemented right now)... Better to do GetAllPods() and then filter them once you have all of them.